tests/build/build_templates.go

```diff
@@ -125,10 +140,28 @@ var _ = framework.BuildSuiteDescribe("Build templates E2E test", Label("build",
 			}
 		})
 
+		It(fmt.Sprintf("triggers PipelineRun for symlink component with source URL %s", pythonComponentGitSourceURL), Label(buildTemplatesTestLabel), func() {
+			timeout := time.Minute * 5
+			Eventually(func() error {
+				pipelineRun, err := kubeadminClient.HasController.GetComponentPipelineRun(symlinkComponentName, applicationName, testNamespace, "")
+				if err != nil {
+					GinkgoWriter.Printf("PipelineRun has not been created yet for symlink Component %s/%s\n", testNamespace, symlinkComponentName)
+					return err
+				}
+				if !pipelineRun.HasStarted() {
+					return fmt.Errorf("pipelinerun %s/%s has not started yet", pipelineRun.GetNamespace(), pipelineRun.GetName())
+				}
+				err = kubeadminClient.TektonController.AddFinalizerToPipelineRun(pipelineRun, finalizerName)
+				if err != nil {
+					return fmt.Errorf("error while adding finalizer: %v", err)
```

Comment:
would be good to mention PipelineRun name in the error message

Reply:
Good point. Will add that.